internal/gatewayapi/tls.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// validateTLSSecretData ensures the cert and key provided in a secret
+// is not malformed and can be properly parsed.
+func validateTLSSecretData(secret *corev1.Secret) error {
+
+	certData := secret.Data[corev1.TLSCertKey]
+
+	certBlock, _ := pem.Decode(certData)
+	if certBlock == nil {
+		return fmt.Errorf("unable to parse pem data in %s", corev1.TLSCertKey)
```

Comment:
dont think we should be logging the key value

Reply:
`corev1.TLSCertKey` is an alias to the key name inside the secret https://github.com/kubernetes/api/blob/master/core/v1/types.go#L6372. 

But moreover, do we want to enforce that secrets be of type `kubernetes.io/tls`? I don't think the conformance test actually enforces this so it would be a restriction we would be apply additionally applying

Reply:
@haq204 I think its okay to start with this strict check since that is the only type we support today, we can loosen it as we start supporting other types which the API supports
https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.GatewayTLSConfig

Reply:
@haq204 - This internal helper function is currently written with the assumption that the secret is valid because it is done in the main validate function as seen here: https://github.com/envoyproxy/gateway/blob/main/internal/gatewayapi/validate.go#L314 

@arkodg - That is just the key so that it gives the user an additional indication of which part of the secret is bad...is the cert or the key. Do you still want me to remove it and make it generic or hard code the key in place?

Reply:
@LanceEa it should be perfectly fine to keep the error strings as is, was a mistake on my end